ocean_lib/data_provider/test/test_data_service_provider.py

```diff
@@ -72,19 +74,34 @@ def test_set_http_client(with_nice_client):
     assert isinstance(DataSP.get_http_client(), HttpClientNiceMock)
 
 
-def test_initialize_fails(with_evil_client):
-    """Tests failure of initialize endpoint."""
+def test_initialize_fails(config):
+    """Tests failures of initialize endpoint."""
+    mock_service = Service(
+        service_id="some_service_id",
+        service_type="some_service_type",
+        service_endpoint="http://mock/",
+        datatoken="some_dt",
+        files="some_files",
+        timeout=0,
+    )
+    with pytest.raises(InvalidURL) as err:
+        DataSP.initialize(
+            "some_did",
+            mock_service,
+            "some_consumer_address",
+            userdata={"test_dict_key": "test_dict_value"},
+        )
+    assert err.value.args[0] == f"InvalidURL {mock_service.service_endpoint}."
```

Comment:
you can also use the match= argument of pytest for brevity. See examples in the compute tests (v4main)